man/bootup.xml

```diff
@@ -92,53 +92,55 @@
-         v                  v                  |                    v                    v
-     (various           (various               |                (various          rescue.service
-    timers...)          paths...)              |               sockets...)               |
-         |                  |                  |                    |                    v
-         v                  v                  |                    v              <emphasis>rescue.target</emphasis>
-   timers.target      paths.target             |             sockets.target
-         |                  |                  |                    |
-         v                  \_________________ | ___________________/
-                                              \|/
-                                               v
-                                         basic.target
-                                               |
-          ____________________________________/|                                 emergency.service
-         /                  |                  |                                         |
-         |                  |                  |                                         v
-         v                  v                  v                                 <emphasis>emergency.target</emphasis>
-     display-        (various system    (various system
- manager.service         services           services)
-         |             required for            |
-         |            graphical UIs)           v
-         |                  |           <emphasis>multi-user.target</emphasis>
-         |                  |                  |
-         \_________________ | _________________/
+<programlisting>cryptsetup-pre.target
+         |      \---------------\                    (various low-level
+         v                      |                     API VFS mounts:
+(local cryptsetup devices) (... and remote)           mqueue, configfs,
+         |                      |                     debugfs, ...)
+         v                      v                         |
+  cryptsetup.target      remote-cryptsetup.target         |
+         |                      |                         |
+         v                      v                         |
+ local-fs-pre.target       remote-fs-pre.target           |
+         |                      |                         |
+         v                      v                         |
+(various mounts and    (network file       (various swap  | (various low-level
+ fsck services...)           systems)       devices...)   |  services: udevd,
+         |                      |                |        |  tmpfiles, random
+         v                      |                v        |  seed, sysctl, ...)
+  local-fs.target       remote-fs.target    swap.target   |          |
+         |                      |                |        |          |
+         \______________________v__________   ___|________|__________/
```

Comment:
this is actually not correct. remote-fs.target is not ordered before sysinit.target. We generally accept remote mounts to happen much later, i.e. after NetworkManager (which traditionally has been a late boot service) and such.

In basic.target we have an explicit dep on var.mount, var-tmp.mount and tmp.mount btw, which pull those three into early boot, in case people have them via NFS or so, but that's only for those three, and that's after sysinit.target too,

The idea is basically: remote stuff might need special deps that are ok to be late boot stuff, thus have much weaker requirements. It's the whole reason local-fs.target and remote-fs.target are two different things actually